### PR TITLE
feat(cli): exclude non-translatable tags inside SVG from requests to LLM

### PR DESCRIPTION
**Pull request by AndreyHirsa**
## Summary

 Moving to a strategy-based approach allows the extractor to make context-aware decisions: elements inside SVG are       
 handled differently than regular HTML elements, and only semantically meaningful text (title, text, desc) is      
 extracted for translation.  

## Changes

 - Introduced `SvgExtractionStrategy` pattern with three modes: `extract`, `skip-recurse`, and `process-normal`                                                                 
 - Added `isInsideSvg` helper to detect SVG context by traversing the DOM tree            
 - Added `containsSvgDescendant` to prevent extracting innerHTML of elements containing nested SVGs                                     
 - Unified `createElementExtractor` as a shared utility between HTML and Twig loaders
 - Moved shared constants (PHRASING_ELEMENTS, BLOCK_ELEMENTS, UNLOCALIZABLE_TAGS) to `element-extraction.ts`
 - Add complex SVG examples to HTML, Twig, and EJS demo files                                                         

## Testing

**Business logic tests added:**

- [x] Unit tests for `getSvgExtractionStrategy` covering all three strategy modes and edge cases (nested SVG, mixed content,  
  context detection)                                                                                                      
- [x] Integration tests for `createElementExtractor` validating extraction, recursion, and attribute handling                 
- [x] End-to-end tests in `html.spec.ts` and `twig.spec.ts` for SVG inside block/phrasing elements, ensuring correct extraction 
  and push/reconstruct behavior 

## Checklist

- [x] Changeset added (if version bump needed)
- [x] Tests cover business logic (not just happy path)
- [ ] No breaking changes (or documented below)

Closes N/A

## Summary by CodeRabbit

* **Chore**
  * Added a changeset entry for a patch release.

* **New Features**
  * Added decorative inline SVG cards/panels and adjacent buttons/icons across demo templates (EN/ES).

* **Bug Fixes**
  * Improved handling of inline SVG content for correct extraction and localization.

* **Changes**
  * Updated forgot-password alert to English and adjusted Spanish bio textarea placeholder.

* **Tests**
  * Added comprehensive tests for HTML/SVG extraction and element-extraction behavior.

* **Refactor**
  * Reorganized extraction logic to be SVG-aware and more reusable.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/cli/src/cli/utils/element-extraction.ts [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+      PHRASING_ELEMENTS.has(tagName) &&
+      hasTranslatableContent(element) &&
+      !containsSvgDescendant(element)
+    ) {
+      const content = context.getInnerHTML(element).trim();
+      if (content) {
+        result[path] = content;
+      }
+      // Don't recurse - innerHTML captures everything
+      return;
```
[the remark this answers is not in the document]
Comment (AndreyHirsa) [reply packages/cli/src/cli/utils/element-extraction.ts:382]: That’s a fair point, but it's not directly related to the current сhanges. In practice, having <script> or <style> inside phrasing elements is an edge case. I’d prefer to keep this PR focused on the SVG extraction logic. @vrcprl

File: packages/cli/demo/ejs/es/example.ejs [hunk 2]
```diff
@@ -49,7 +72,7 @@
     
     <script>
         function showForgotPassword() {
-            alert("Contacta con soporte para obtener ayuda.");
+            alert("Contact support for help.");
         }
```
[the remark this answers is not in the document]
Comment (AndreyHirsa) [reply packages/cli/demo/ejs/es/example.ejs:76]: @vrcprl It looks like content inside <script> is being translated in the EJS loader. That’s definitely not the expected behavior. Should we open a ticket to track this?

File: packages/cli/src/cli/utils/element-extraction.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vrcprl) [reply packages/cli/src/cli/utils/element-extraction.ts:382]: Looks good, let’s keep it as is, no ticket needed.
